src/controllers/reservation.controller.ts

```diff
@@ -0,0 +1,159 @@
+import moment from "moment";
+import {
+  CreateReservationInput,
+  RequestCancellationParams,
+  RequestCancellationInput,
+  DeleteRequestParams,
+  DeleteRequestInput,
+} from "./../schemes/reservation.schema";
+import { Request, Response } from "express";
+import {
+  findReservation,
+  createReservation,
+  findReservationsOnGivenDate,
+  findReservationById,
+  deleteReservationById
+} from "../services/reservation.service";
+import { findTableBySeatNumber } from "../services/table.service";
+import sendEmail from "../utils/mailer";
+import log from "../utils/logger";
+import { customAlphabet } from "nanoid";
+export async function createReservationHandler(
+  req: Request<{}, {}, CreateReservationInput>,
+  res: Response
+) {
+  const { date, email, seatNumber, numberOfSeats, duration, fullName } =
+    req.body;
+
+  const reservationExists = await findReservation({
+    date,
+    duration,
+    seatNumber,
+  });
+  const table = await findTableBySeatNumber(seatNumber);
+
+  if (reservationExists)
+    return res
+      .status(400)
+      .json({ error: "This table is alredy taken in this time" });
+  if (!table)
+    return res.status(404).send("Table of given number does not exist");
+
+  if (numberOfSeats > table.maxNumberOfSeats)
```

Comment:
You should always use full if structure, especially when the body is on several lines